concourse/pipelines/create-bosh-cloudfoundry.yml

```diff
@@ -705,6 +747,111 @@ jobs:
+              FIREHOSE_USER=graphite-nozzle
+              FIREHOSE_PASS=$($VAL_FROM_YAML secrets.uaa_clients_firehose_password cf-secrets/cf-secrets.yml)
+              API_ENDPOINT=$($VAL_FROM_YAML properties.cc.srv_api_uri cf-manifest/cf-manifest.yml)
+              UAA_ENDPOINT=$($VAL_FROM_YAML properties.uaa.url cf-manifest/cf-manifest.yml)
+              DOPPLER_ENDPOINT=$($VAL_FROM_YAML properties.loggregator.traffic_controller_url cf-manifest/cf-manifest.yml)
+              GRAPHITE_SERVER=$($VAL_FROM_YAML jobs.graphite_z1.networks.cf1.static_ips.0 cf-manifest/cf-manifest.yml)
+
+              for var_name in CF_ADMIN CF_PASS FIREHOSE_USER FIREHOSE_PASS API_ENDPOINT UAA_ENDPOINT DOPPLER_ENDPOINT GRAPHITE_SERVER; do
+                echo export "${var_name}"=\"$(eval echo \$${var_name})\"
+              done > config/config.sh
+
+    - task: deploy-graphite-nozzle
+      config:
+        platform: linux
+        inputs:
+          - name: graphite-nozzle
+          - name: config
+        params:
+        # FIXME: use docker:///governmentpaas/cf-cli when is ready
+        image: docker:///keymon/docker-cf-cli
+        run:
+          path: sh
+          args:
+            - -e
+            - -c
+            - |
+              . ./config/config.sh
+
+              cf api --skip-ssl-validation ${API_ENDPOINT}
+              echo | cf login -u ${CF_ADMIN} -p ${CF_PASS}
+              echo | cf create-org admin
+              cf create-space admin -o admin
+              cf target -o admin -s admin
+
+              cat <<EOF > graphite-nozzle.json
+              [
+                  {"protocol":"udp","destination":"${GRAPHITE_SERVER}","ports":"8125"}
+              ]
+              EOF
+              cf create-security-group graphite-nozzle graphite-nozzle.json
+              cf bind-staging-security-group graphite-nozzle
+              cf bind-running-security-group graphite-nozzle
+
+              cd graphite-nozzle
+
+              # The sleep 1 is to avoid logs being eaten if the configuration is wrong
+              # due to this bug: https://github.com/cloudfoundry/loggregator/issues/86
+              echo "web: sleep 1 && graphite-nozzle" > Procfile
+
+              cat <<EOF > manifest.yml
+              ---
+              applications:
+              - name: graphite-nozzle
+                memory: 100M
+                instances: 1
+                no-route: true
+                health-check-type: none
+                env:
+                  DOPPLER_ENDPOINT: "${DOPPLER_ENDPOINT}"
+                  UAA_ENDPOINT: "${UAA_ENDPOINT}"
+                  STATSD_ENDPOINT: "${GRAPHITE_SERVER}:8125"
+                  FIREHOSE_USERNAME: "${FIREHOSE_USER}"
+                  FIREHOSE_PASSWORD: "${FIREHOSE_PASS}"
+                  SUBSCRIPTION_ID: "firehose"
+                  STATSD_PREFIX: "cfstats."
+                  SKIP_SSL_VALIDATION: "true"
+                  PREFIX_JOB: "true"
+              EOF
+
+              cf push
```

Comment:
/cc @Jonty 
> Rather than cf-pushing on every run, we could use something like cf apps | grep graphite-nozzle to only push the app if it's not already deployed? It's really not good to be pushing on every deploy as it'll restart the app and we'll have metric gaps.

Yes, but that won't update the app if there is an update. Other approach would be put the git hash in the name of the app, and delete the old one if required. 

But as we comment in the commit, this is a temporary solution, we will deploy this app with a release. Unless you are strongly against it, I suggest to leave it as it is.